arch/x86/core/fatal.c

```diff
@@ -52,7 +52,7 @@ bool z_x86_check_stack_bounds(uintptr_t addr, size_t size, uint16_t cs)
 	if (_current == NULL || arch_is_in_isr()) {
 		/* We were servicing an interrupt or in early boot environment
 		 * and are supposed to be on the interrupt stack */
-		int cpu_id;
+		uint8_t cpu_id;
```

Comment:
Again confused.  It's true that the "id" field of struct _cpu is a byte, but that's an implementation detail.   Code working with it (which is rare, and limited to low level drivers like this) wants to be using a machine word for portability and simplicity.